sunpy/map/mapbase.py

```diff
@@ -1941,19 +1941,16 @@ def draw_rectangle(self, bottom_left, *, width: u.deg = None, height: u.deg = No
         Extra keyword arguments to this function are passed through to the
         `~matplotlib.patches.Rectangle` instance.
         """
-        if isinstance(top_right, u.Quantity) and isinstance(width, u.Quantity):
-            # The decorator assigns the first positional arg to top_right and so on.
-            height = width
-            width = top_right
-            top_right = None
-
         bottom_left, top_right = get_rectangle_coordinates(bottom_left,
                                                            top_right=top_right,
                                                            width=width,
                                                            height=height)
 
+        bottom_left = bottom_left.transform_to(self.coordinate_frame)
+        top_right = top_right.transform_to(self.coordinate_frame)
```

Comment:
Not for this PR, but I wonder if it would be worth adding a `coordinate_frame` argument to `get_rectangle_coordinates` so the conversion is automatically handled in that function.

Reply:
I think that over-complicates `get_rectangle_coordinates()`.  Ideally, `get_rectangle_coordinates()` would return a 2-element `SkyCoord` rather than a tuple.  Then, a subsequent `.transform_to()` call would be a single line.